src/backend/catalog/pg_constraint.c

```diff
@@ -836,6 +836,59 @@ get_constraint_relation_oids(Oid constraint_oid, Oid *conrelid, Oid *confrelid)
 	ReleaseSysCache(tup);
 }
 
+/*
+ * get_constraint_relation_columns
+ *		Find the columns of the relations to which a constraint refers.
+ *		Returns the list of found columns.
+ */
+List *
+get_constraint_relation_columns(Oid constraint_oid)
```

Comment:
Doesn't the `gpdb::GetRelationKeys` function do the same thing?

Reply:
gpdb::GetRelationKeys takes relation OID as a parameter, and may return several keysets (in case there are CONSTRAINT_UNIQUE constraints). But it looks that we will not be able to filter out UNIQUE keysets that it might return, which we shouldn't take into account. So we have to retrieve key set basing on OID of primary key constraint.

Reply:
Maybe then you can rewrite your `get_constraint_relation_columns` function similar to the `get_relation_keys` function? For example, using `deconstruct_array` and so on.